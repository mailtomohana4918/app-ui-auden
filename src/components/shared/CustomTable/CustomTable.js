import React,{useState} from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
  Table,
  TablePagination, Typography, Grid
} from "@material-ui/core";
import { withRouter } from "react-router-dom";


const useStyles = makeStyles(() => ({
  root: { width: "100%" },
  typoGraphy:{paddingBottom:"20px"},
  table:{minWidth:"100"}
}));

const CustomTable = (props) => {
  const classes = useStyles();
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(10);

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(+event.target.value);
    setPage(0);
  };

  const handleRowClick = (id, source) => {
    source === "repo" && props.history.push(`/issue/search/${id}`);
  };

  return (
    <>
     <Grid
        container
        spacing={0}
        direction="column"
        alignItems="center"
        justify="center"
      >
    <Typography className={classes.typoGraphy} variant="h5" color="primary"> {props.source==="repo" ? "Repository Details":"Issue Details"}</Typography>
      <Paper className={classes.root}>
        <TableContainer className={classes.container}>
          <Table className={classes.table} stickyHeader aria-label="sticky table">
            <TableHead>
              <TableRow>
                {props.columns.map((column) => (
                  <TableCell
                    key={column.id}
                    align={column.align}
                    style={{ minWidth: column.minWidth }}
                  >
                    {column.label}
                  </TableCell>
                ))}
              </TableRow>
            </TableHead>
            <TableBody>
              {props.searchResult
                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                .map((row) => {
                  return (
                    <TableRow hover role="checkbox" tabIndex={-1} key={row.id}>
                      {props.columns.map((column) => {
                        const value = row[column.id];
                        return (
                          <TableCell
                            key={column.id}
                            align={column.align}
                            onClick={() => handleRowClick(row.id, props.source)}
                          >
                            {column.format && typeof value === "number"
                              ? column.format(value)
                              : value}
                          </TableCell>
                        );
                      })}
                    </TableRow>
                  );
                })}
            </TableBody>
          </Table>
        </TableContainer>
        <TablePagination
          rowsPerPageOptions={[10, 25, 100]}
          component="div"
          count={props.searchResult.length}
          rowsPerPage={rowsPerPage}
          page={page}
          onChangePage={handleChangePage}
          onChangeRowsPerPage={handleChangeRowsPerPage}
        />
      </Paper>
      </Grid>
    </>
  );
};
export default withRouter(CustomTable);
