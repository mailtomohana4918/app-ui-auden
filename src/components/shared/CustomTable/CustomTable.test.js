import { render } from "@testing-library/react";
import CustomTable from "./CustomTable";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
describe("CustomTable Repository details", () => {
  const columns = [
    { id: "name", label: "Name", minWidth: 100 },
    {
      id: "watchers",
      label: "Stars",
      minWidth: 170,
      align: "right",
      format: (value) => value.toLocaleString("en-US"),
    },
    {
      id: "forks",
      label: "Forks",
      minWidth: 170,
      align: "right",
      format: (value) => value,
    },
    {
      id: "open_issues",
      label: "Issues",
      minWidth: 170,
      align: "right",
      format: (value) => value,
    },
  ];

  const data = [
    { id: 1, name: "React", watchers: 2000, forks: 100, open_issues: 200 },

    { id: 2, name: "Vue", watchers: 3000, forks: 200, open_issues: 300 },
    { id: 3, name: "Angular", watchers: 4000, forks: 300, open_issues: 400 },
    { id: 4, name: "Django", watchers: 5000, forks: 400, open_issues: 500 },
  ];
  it("render component when source is repo", () => {
    const { asFragment } = render(
      <Router>
        <CustomTable source="repo" columns={columns} searchResult={data} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("CustomTable Issue details", () => {
  const columns = [
    { id: "title", label: "Issue", minWidth: 100 },
    { id: "number", label: "Issue Number", minWidth: 100 },
    {
      id: "state",
      label: "Status",
      minWidth: 170,
      align: "right",
      format: (value) => value.toLocaleString("en-US"),
    },
    {
      id: "created_at",
      label: "Created Date",
      minWidth: 170,
      align: "right",
      format: (value) => value,
    },
    {
      id: "updated_at",
      label: "Last Updated Date",
      minWidth: 170,
      align: "right",
      format: (value) => value,
    },
  ];

  const issue_Date = "2020-07-27T16:48:02Z";
  const data = [
    {
      id: 1,
      title: "Component re-render",
      state: 2000,
      created_at: issue_Date,
      updated_at: issue_Date,
    },

    {
      id: 2,
      title: "Chaining Issue",
      state: 2000,
      created_at: issue_Date,
      updated_at: issue_Date,
    },
    {
      id: 3,
      title: "Media query not rendering",
      state: 2000,
      created_at: issue_Date,
      updated_at: issue_Date,
    },
    {
      id: 4,
      title: "snapshots not updating",
      state: 2000,
      created_at: issue_Date,
      updated_at: issue_Date,
    },
  ];
  it("render component when source is issue", () => {
    const { asFragment } = render(
      <Router>
        <CustomTable columns={columns} searchResult={data} source="issue"/>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
