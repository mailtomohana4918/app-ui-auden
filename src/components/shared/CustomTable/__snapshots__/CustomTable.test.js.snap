// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTable Issue details render component when source is issue 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <h5
      class="MuiTypography-root makeStyles-typoGraphy-5 MuiTypography-h5 MuiTypography-colorPrimary"
    >
       Issue Details
    </h5>
    <div
      class="MuiPaper-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiTableContainer-root"
      >
        <table
          aria-label="sticky table"
          class="MuiTable-root makeStyles-table-6 MuiTable-stickyHeader"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Issue
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Issue Number
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 170px;"
              >
                Status
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 170px;"
              >
                Created Date
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 170px;"
              >
                Last Updated Date
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Component re-render
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Chaining Issue
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Media query not rendering
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                snapshots not updating
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2020-07-27T16:48:02Z
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            id="mui-12810"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-12810 mui-28948"
              class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              id="mui-28948"
              role="button"
              tabindex="0"
            >
              10
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            1-4 of 4
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Previous page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Next page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CustomTable Repository details render component when source is repo 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <h5
      class="MuiTypography-root makeStyles-typoGraphy-2 MuiTypography-h5 MuiTypography-colorPrimary"
    >
       Repository Details
    </h5>
    <div
      class="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiTableContainer-root"
      >
        <table
          aria-label="sticky table"
          class="MuiTable-root makeStyles-table-3 MuiTable-stickyHeader"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 170px;"
              >
                Stars
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 170px;"
              >
                Forks
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 170px;"
              >
                Issues
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                React
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                100
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                200
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Vue
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                3,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                200
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                300
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Angular
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                4,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                300
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                400
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Django
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                5,000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                400
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                500
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            id="mui-71973"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-71973 mui-103"
              class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              id="mui-103"
              role="button"
              tabindex="0"
            >
              10
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            1-4 of 4
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Previous page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Next page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
